xpartamupp/xpartamupp.py

```diff
@@ -75,22 +82,48 @@ def remove_game(self, jid):
 
         """
         try:
-            del self.games[jid]
+            del self.games[jid.bare][jid.resource]
+            if len(self.games[jid.bare]) == 0:
+                del self.games[jid.bare]
         except KeyError:
             logging.warning("Game for jid %s didn't exist", jid)
             return False
         else:
             return True
 
+    def remove_all_games_for_jid(self, jid):
+        """Remove all games attached to the bare version of the JID.
+
+        Arguments:
+            jid (sleekxmpp.jid.JID): JID of the player whose games to
+                remove.
+
+        Returns:
+            True if removing the games succeeded, False if not
+
+        """
+        try:
+            del self.games[jid.bare]
+        except KeyError:
+            logging.warning("No games for jid %s exist", jid)
+            return False
+        else:
+            return True
+
     def get_all_games(self):
         """Return all games.
 
         Returns:
-            dict containing all games with the JID of the player who
-            started the game as key.
+            list containing all games. Sorted by jid then by resource.
 
         """
-        return self.games
+        return [game
+                for jid, dict_of_resources_and_games
+                in sorted(self.games.items(),
+                          key=lambda item: item[0])         # sort by jid
+                for resource, game
+                in sorted(dict_of_resources_and_games.items(),
+                          key=lambda item: item[0])]        # sort by resource
```

Comment:
I find this statement pretty hard to understand. If I'm not mistaken the given sort key is also not necessary, as it results in the same behavior than using the default sort key. That'd allow writing it down in a way I find much more pleasing:
```suggestion
        games = []
        for _, games_for_jid in sorted(self.games.items()):
            for _, game in sorted(games_for_jid.items()):
                games.append(game)
        return games
```

I'm also not sure if the lambda function you provided as sort key works with versions of Python older than 3.6 so unless you specifically tested that I suggest merging https://github.com/0ad/lobby-bots/pull/4 first, just to be sure not to break something.